### ISSUE TITLE
Retrieve schema information from DB connection

### ISSUE THREAD

**Issue by phoenix-tr** [DarkWanderer/ClickHouse.Client#20]
We need to retrieve table columns' information from DB connection. Currently we are getting it from DB connection interfaces (IDBConnection) by using GetSchema interface but for Clickhouse it gives the following error;

> System.NotSupportedException: 'Specified method is not supported.'

For Oracle it returns related column information with their data types. I need to use same GetSchema method for Clickhouse connection (ClickHouseConnection) as well. It returns System.Data.DataTable object which we are using for data type differentiation for each coming column name.

As a workaround we wrote our own query to get them like below;

`SELECT` name ColumnName, CASE WHEN type LIKE ('%Int%') OR type LIKE ('%Float%') OR type LIKE ('%Decimal%') THEN 'NUMBER' WHEN type LIKE ('%Date%') THEN 'DATE' ELSE 'STRING' END DataType FROM system.columns WHERE database = '{schemaName}' and table = '{tableNameWithoutSchema}'

So basically GetSchema could return all of the columns within a System.Data.DataRow object for each row in System.Data.DataTable object.

**Comment by DarkWanderer** [comment]
Aliases like `NUMBER` or `STRING` are Oracle-specific, so I won't really be able to support them. However, I am able to provide information about:

1. ClickHouse type
2. Short form of corresponding framework type

Database | Table | Name | ProviderType | DataType
-- | -- | -- | -- | --
system | functions | name | String | String
system | functions | is_aggregate | UInt8 | Byte
system | functions | case_insensitive | UInt8 | Byte
system | functions | alias_to | String | String

This is currently in `master` - please check it out and let me know your thoughts.

P.S. Please keep in mind that order of columns may change, and new columns might be added - best to refer to columns by name.

**Comment by phoenix-tr** [comment]
Those aliases are just for the example of our usage. I think that schema info should return exact DB types from database like Oracle. If anyone wants to take them as aliases they can evaluate by themselves. I put the code like this because I am merging the data from both CH and Ora together to handle in my application.

**Comment by DarkWanderer** [comment]
ProviderType column in above example returns string with exact ClickHouse type, if you need it

**Comment by siradjev** [comment]
Thank you @DarkWanderer.  For decimals it's returning Decimal(x,y), can it return actual type such as Decimal32/64/128?

**Comment by DarkWanderer** [comment]
Curiously, `Decimal(x,y)` is what's actually stored in `system.columns` table. I have no problem converting that to `Decimal(32|64|128)`, however - will be in for next release

**Comment by siradjev** [comment]
> Curiously, `Decimal(x,y)` is what's actually stored in `system.columns` table. I have no problem converting that to `Decimal(32|64|128)`, however - will be in for next release

I did not think of that. Do you think it would also make more sense to ask clickhouse developers to introduce toDecimal(value,precision,scale) function, then this implementation can be fixed back to returning consistent datatype with the clickhouse?

**Comment by DarkWanderer** [comment]
ClickHouse devs would be a better authority, but from what I know, CH uses strict typing system and all functions must have well-defined return type. A function like `toDecimal` would not be possible, as it would have to be able to return types `Decimal32`/`Decimal64`/`Decimal128` depending on input

**Comment by DarkWanderer** [comment]
Released, please reopen in case of problems with latest version
